### PR TITLE
check MSRV in CI, lower MSRV to 1.57.0

### LINKED ISSUE #2460: master seems broken: error[E0277]: `ErrorKind` doesn't implement `std::fmt::Display`
### Summary

from `cargo install --path helix-term`, latest commit at date: `7b287f63 Blaž Hrastnik 13 hours ago - nix: update flake dependencies`

```
   Compiling helix-view v0.6.0 (/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/helix-view)
error[E0277]: `ErrorKind` doesn't implement `std::fmt::Display`
   --> helix-view/src/handlers/dap.rs:305:81
    |
305 | ...                   e => panic!("Error to start debug console: {}", e),
    |                                                                       ^ `ErrorKind` cannot be formatted with the default formatter
    |
    = help: the trait `std::fmt::Display` is not implemented for `ErrorKind`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `$crate::const_format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `helix-view` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `helix-term v0.6.0 (/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/helix-term)`, intermediate artifacts can be found at `/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/target`
```

### Reproduction Steps

 

I tried this:

1. `cargo install --path helix-term`

I expected this to happen:

Compile and install

Instead, this happened:

```
   Compiling helix-view v0.6.0 (/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/helix-view)
error[E0277]: `ErrorKind` doesn't implement `std::fmt::Display`
   --> helix-view/src/handlers/dap.rs:305:81
    |
305 | ...                   e => panic!("Error to start debug console: {}", e),
    |                                                                       ^ `ErrorKind` cannot be formatted with the default formatter
    |
    = help: the trait `std::fmt::Display` is not implemented for `ErrorKind`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `$crate::const_format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `helix-view` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `helix-term v0.6.0 (/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/helix-term)`, intermediate artifacts can be found at `/Users/juan.alvarez/Library/Caches/Homebrew/helix--git/target`
```

### Helix log

_No response_

### Platform

macOS

### Terminal Emulator

iTern2

### Helix Version

22.03-207-g7b287f63

### LINKED ISSUE #2482: Check MSRV in CI
It's easy to accidentally bump the MSRV by using a newer Rust feature. We might be able to check that we're adhering to MSRV with `cargo msrv verify` in a job in the CI build.

For background: we keep an MSRV that is behind a few versions in order to make packaging easier for some package managers - some package managers have long processes for updating rust versions and using latest stable Rust features may make it impossible to build Helix for those package managers.

See also: https://github.com/helix-editor/helix/pull/2475#discussion_r873261249

### PR DESCRIPTION
**Pull request by the-mikedavis**
It's very easy to use new rust features without realizing it since
the CI and local development workflows may use the latest rust version.
We try to keep some backwards compatibility with rust versions to make
packaging easier for some OS-level package-managers like Void Linux's.
See #1881.

This change adds a new CI check which uses the cargo-msrv crate to
verify an MSRV version recorded in helix-term's Cargo.toml. When
changing MSRV, we can then edit `helix-term/Cargo.toml`'s
`package.metadata.msrv` key.

The MSRV is set at 1.60.0 for the sake of checkmarks on builds but
ideally we want to support 1.57.0. The PR that fixes the MSRV in #2460
should set this MSRV metadata key back to 1.57.0.

Closes #2482.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -48,6 +48,53 @@ jobs:
...
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-msrv
+
+      - name: Run cargo check
+        uses: actions-rs/cargo@v1
+        with:
+          command: msrv
+          args: --verify --path=helix-term
```
Comment (pickfire) [inline .github/workflows/build.yml:96]: Is this still `cargo check`?

Comment (pickfire) [inline .github/workflows/build.yml:96]: Are these really cargo check, not cargo msrv?

Comment (the-mikedavis) [reply .github/workflows/build.yml:96]: Whoops, you caught me copy-pasting 😅

**Comment by the-mikedavis** [comment]
Actually I think it's best to do this by using a matrix for the Check job: `cargo-msrv` doesn't say what failed the check. Plus it seems to misbehave on 1.57.0 🤔

**Pull request closed without merging**